### DISCUSSION TITLE
ref in template typescript error?

### DISCUSSION THREAD

**Discussion by lovetingyuan** [vuejs/language-tools#1003] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/7310471/156365417-10be4154-00fe-4286-b5cd-cec7134134b7.png)
Ref type will auto unwrap in template, item.prop type is string, why this not work?
```log
error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Ref<{ [x: string]: any; }>'.
  No index signature with a parameter of type 'string' was found on type 'Ref<{ [x: string]: any; }>'.

18             :data.sync="formData[item.prop]"
                           ~~~~~~~~~~~~~~~~~~~
```

**Comment by xiaoxiangmoe** [comment, marked as answer]
Can you provide a minimal reproducible demo?

---

Maybe the `Ref<T>` type of formData and the `Ref<T>` type from vue is not same.

Try this:

```ts
import {type Ref} from 'vue'
// ...
const formData2: Ref<{[x:string]: any}> = formData;
```

If type is not same, this will throw error.

**Reply by lovetingyuan** [reply]
I can not reproduce this issue, I will close this. Thank you.
